Fix EmptyLinesAroundMethodBody for methods with arguments spanning multiple lines

aduth opened the pull request:
Fixes #13430

Fixes `Layout/EmptyLinesAroundMethodBody` to identify empty lines for methods which have arguments spanning multiple lines.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

dvandersluis commented:
Hi @aduth, thanks for your contribution! Can you please make sure that the PR checklist is retained and filled out? I've added it back in here but please go through it.

dvandersluis commented on changelog/fix_fix_empty_lines_around_method_body_for_methods.md:
```diff
@@ -0,0 +1 @@
+* [#13430](https://github.com/rubocop/rubocop/pull/13430): Fix EmptyLinesAroundMethodBody for methods with arguments spanning multiple lines. ([@aduth][])
```
```suggestion
* [#13430](https://github.com/rubocop/rubocop/issues/13430): Fix EmptyLinesAroundMethodBody for methods with arguments spanning multiple lines. ([@aduth][])
```

aduth replied:
Whoops, I was trying to be clever about getting the "next" PR number in here ahead of actually creating the pull request, but my issue took the number I was expecting for the PR.

Would you prefer that this point to the issue #13430 ?

```suggestion
* [#13430](https://github.com/rubocop/rubocop/issues/13430): Fix EmptyLinesAroundMethodBody for methods with arguments spanning multiple lines. ([@aduth][])
```

...or the pull request #13431 ?

```suggestion
* [#13431](https://github.com/rubocop/rubocop/pull/13431): Fix EmptyLinesAroundMethodBody for methods with arguments spanning multiple lines. ([@aduth][])
```

dvandersluis replied:
When there's an issue, we link the changelog to the issue, not the PR. Only if there is only a PR without an issue do we link to the PR.

aduth replied:
> When there's an issue, we link the changelog to the issue, not the PR. Only if there is only a PR without an issue do we link to the PR.

Makes sense 👍 Updated in e5b58d0.

dvandersluis commented:
There's another false negative case, would you like to take a look at it?

```ruby
def foo(

)

  # ...
end
```

aduth commented:
> Hi @aduth, thanks for your contribution! Can you please make sure that the PR checklist is retained and filled out? I've added it back in here but please go through it.

Thanks, I thought I'd ran through them all and left it out as implied completed, but I must have missed `bundle exec rake default`. Ran that, fixed the issues in 43403eb, and updated to check the items in the original comment.

aduth commented:
>There's another false negative case, would you like to take a look at it?

I'll try to take a shot at it! Good catch

dvandersluis commented:
Thanks! Can you please squash your commits together?

aduth commented:
> Thanks! Can you please squash your commits together?

Done 👍 Squashed to 85f1eed

dvandersluis commented:
Thank you!
